internal/controller/providers/psmdb/applier.go

```diff
@@ -54,9 +54,20 @@ func (p *applier) Paused(paused bool) {
 
 //nolint:staticcheck //using deprecated field for backward compatibility
 func (p *applier) AllowUnsafeConfig() {
-	// 1-node cluster, former 1-node cluster, a cluster with deprecated AllowUnsafeConfiguration
-	value := p.DB.Spec.Engine.Replicas == 1 || p.PerconaServerMongoDB.Spec.UnsafeConf || p.DB.Spec.AllowUnsafeConfiguration
-	p.PerconaServerMongoDB.Spec.UnsafeConf = value
+	p.PerconaServerMongoDB.Spec.UnsafeConf = false
+	useInsecureSize := p.DB.Spec.Engine.Replicas == 1 || p.DB.Spec.AllowUnsafeConfiguration
+	p.PerconaServerMongoDB.Spec.Unsafe = psmdbv1.UnsafeFlags{
+		TLS:                    p.DB.Spec.AllowUnsafeConfiguration,
+		MongosSize:             useInsecureSize,
+		ReplsetSize:            useInsecureSize,
+		TerminationGracePeriod: p.DB.Spec.AllowUnsafeConfiguration,
+		BackupIfUnhealthy:      p.DB.Spec.AllowUnsafeConfiguration,
+	}
+	if p.DB.Spec.AllowUnsafeConfiguration {
+		p.PerconaServerMongoDB.Spec.TLS = &psmdbv1.TLSSpec{
+			Mode: psmdbv1.TLSModeDisabled,
+		}
+	}
```

Comment:
Using `p.DB.Spec.AllowUnsafeConfiguration` for backward compatibility with the already existing clusters on the Everest side. However on the psmdb side we don't use `p.PerconaServerMongoDB.Spec.UnsafeConf` anymore but use more specific `psmdbv1.UnsafeFlags`  flags. 

[Here](https://github.com/percona/percona-server-mongodb-operator/blob/4d390e49486dbcb244722eb295a35fb0accc77c5/pkg/apis/psmdb/v1/psmdb_defaults.go#L115-L124) is how psmdb handles `psmdbv1.UnsafeFlags`, that's why to make the old clusters work correctly we also need to set `.PerconaServerMongoDB.Spec.TLS` for them.